### PR TITLE
Fix compatibility with GHC-8.4 to 8.10

### PR DESCRIPTION
**Pull request by sjakobi**
Context: #55

### CODE REVIEW

**Comment by sjakobi** [comment]
Travis CI doesn't report the build status, so here's the build job for this PR: https://www.travis-ci.org/github/fmthoma/vgrep/builds/730904099

To fix the Travis settings, maybe try the instructions in https://github.com/quchen/prettyprinter/issues/172#issuecomment-653885208.

File: src/Vgrep/Environment/Config/Sources/File.hs [hunk 1]
```diff
@@ -333,7 +333,7 @@ mapMKeys f = fmap M.fromList . M.foldrWithKey go (pure [])
         xs <- mxs
         pure ((k', x) : xs)
 
-parseChord :: Monad m => String -> m Key.Chord
+parseChord :: MonadFail m => String -> m Key.Chord
```
Comment (sjakobi) [inline src/Vgrep/Environment/Config/Sources/File.hs:336]: This needs a bit of CPP for backward compatibility.

**Comment by sjakobi** [comment]
@fmthoma Could you review? :)

File: src/Vgrep/Environment/Config/Sources/File.hs (hunk 1 above)
Comment (fmthoma) [reply src/Vgrep/Environment/Config/Sources/File.hs:338]: 4.9.0 dates back to 2016 (GHC 8.0.1), so I'm a bit inclined to drop support for GHC 7.x. after almost five years… What do you think?

**Review by fmthoma — CHANGES_REQUESTED** [review]

**Comment by fmthoma** [comment]
Thanks for your PR @sjakobi, and sorry for taking so long to answer! I'll look into Travis etc.

File: src/Vgrep/Environment/Config/Sources/File.hs (hunk 1 above)
Comment (fmthoma) [reply src/Vgrep/Environment/Config/Sources/File.hs:338]: And BTW, I think this is missing an `import Control.Monad.Fail (MonadFail)`

Comment (sjakobi) [reply src/Vgrep/Environment/Config/Sources/File.hs:337]: Removed the CPP and fixed the build.

**Comment by sjakobi** [comment]
> Thanks for your PR @sjakobi, and sorry for taking so long to answer! I'll look into Travis etc.

No worries.

Regarding CI: With Travis limiting the build minutes even for open source projects, you could consider using GitHub actions instead. Please do use some CI system though. Checking GHC compatibility across multiple versions manually is rather annoying.

**Comment by fmthoma** [comment]
Alright, I'll merge and fix the Travis configuration :)

**Pull request merged**